documentation/production/tutorial-production-mode-basic.asciidoc

```diff
@@ -84,3 +84,102 @@ When minifying code, it is often obscured, which makes it harder to read, hence
 Bundling is an optimization where we merge multiple files to a single collection so that the browser doesn't need to
 request so many files from the server. 
 This makes the application load faster.
+
+== Checklist for the production mode app artifact
+
+Vaadin Servlet configures itself from the `flow-build-info.json` file, which is located in the `META-INF/VAADIN/config/` resource package.
+The resources location differ for different artifact types, for example:
+
+. Spring Boot jar file places resources into the `BOOT-INF/classes/` folder in the jar file;
+. WAR archive places resources into the `WEB-INF/classes/` folder.
+
+If the production mode has been activated properly, the contents of the `flow-build-info.json` should be as follows:
+
+.flow-build-info.json
+[source, json]
+----
+{
+  "compatibilityMode": false,
+  "productionMode": true,
+  "enableDevServer": false,
+  "chunks": {
+    "fallback": {
+      "jsModules": [
+        "@vaadin/vaadin-icons/vaadin-icons.js",
+        "@vaadin/vaadin-grid/src/vaadin-grid-tree-toggle.js",
+// etc etc
+        "frontend://ironListConnector.js"
+      ],
+      "cssImports": [
+      ]
+    }
+  }
+}
+----
+
+(Note that the `productionMode` property is set to `true`).
+
+It's very important to have this file on classpath. If it's not, Vaadin Servlet
+will then try to auto-configure itself and sometimes wrongly so. For example when launching
+WAR in Tomcat from Intellij, the `frontendFolder` setting will be "auto-detected"
+to Tomcat's `bin/` folder, which will cause webpack to fail later on, because of missing files.
+
+If the file is present multiple times on the classpath,
+then Vaadin tries to choose which one is the right one (which one is coming
+from the main project). That's unfortunate since sometimes the algorithm fails and
+backfires unexpectedly, choosing the incorrect file.
+This usually happens when a Vaadin addon
+incorrectly includes the `flow-build-info.json` file in the jar file.
+That's a bug in the addon packaging which need to be fixed and the `flow-build-info.json` file removed from their jar files.
+
+For the reasons stated above, keep in mind to have the `flow-build-info.json` file on the classpath exactly once,
+and always coming from your main project.
+
+The project `flow-build-info.json` file is generated by Vaadin plugin, in the `prepare-frontend` task,
+therefore it's very important to always run that task before launching your app in development mode.
```

Comment:
This is not true since it is done by the `flow-server` anyway if it is not there. So basically running the app with `spring-boot:run` or `jetty:run` or via `Application` class in IDE for Spring Boot project works without running `mvn install` before. There has been issues with that at certain points, but not at the moment.

Reply:
There are still issues:
```
	Caused by: java.lang.IllegalStateException: Failed to determine project directory for dev mode. Directory '/home/mavi/local/apache-tomcat-9.0.31/bin' does not look like a Maven or Gradle project. Ensure that you have run the prepare-frontend Maven goal, which generates 'flow-build-info.json', prior to deploying your application
		at com.vaadin.flow.server.startup.DevModeInitializer.getBaseDirectoryFallback(DevModeInitializer.java:377)
		at com.vaadin.flow.server.startup.DevModeInitializer.initDevModeHandler(DevModeInitializer.java:252)
```
however luckily Vaadin will now fail eagerly with informative error message. This is Intellij+Tomcat.

However, you have a point that the information needs to be updated. Please let me update this paragraph.

Reply:
I think for production mode it is needed, but @caalador can please clarify. My point was about development mode where it is not only created by running `prepare-frontend`

Reply:
There might be a slim chance that we could run production-mode without the info file, but I would expect it to fail the build as the fallback chunks are written there during build so I would say it's needed for a production build.

Reply:
You're right - the current text is a mixture of production and development modes. I'll remove all references to the development mode, which should fix this particular item.